Source/Diagnostics/BoundaryScrapingDiagnostics.cpp

```diff
@@ -133,35 +119,47 @@ BoundaryScrapingDiagnostics::DoComputeAndPack (int /*step*/, bool /*force_flush*
 }
 
 bool
-BoundaryScrapingDiagnostics::DoDump (int /*step*/, int /*i_buffer*/, bool force_flush)
+BoundaryScrapingDiagnostics::DoDump (int step, int /*i_buffer*/, bool force_flush)
 {
     if (force_flush) {
         return true;
     } else {
-        return false;
+        return (m_intervals.contains(step+1));
     }
 }
 
 void
 BoundaryScrapingDiagnostics::Flush (int i_buffer)
 {
     auto & warpx = WarpX::GetInstance();
+    ParticleBoundaryBuffer& particle_buffer = warpx.GetParticleBoundaryBuffer();
+
+    int n_particles = 0;
+    for (auto const& species_name : m_output_species_names) {
+        n_particles += particle_buffer.getNumParticlesInContainer(species_name, i_buffer);
+    }
 
-    // This is not a backtransform diagnostics, but we still set the flag `isBTD`
-    // This enables:
-    //   - writing the data that was accumulated in a PinnedMemoryParticleContainer
-    //   - writing repeatedly to the same file
-    bool const isBTD = true;
-    // For now, because this function is currently only called at the very end
-    // of the simulation for BoundaryScrapingDiagnostics, we always set `isLastBTD`.
-    // This tells WarpX to write all the metadata (and not purely the particle data)
-    bool const isLastBTD = true;
+    // If the saving of the particles was not set up for any of the species for this boundary
+    // or if no particles have been lost, then don't write anything out.
+    if (n_particles == 0) return;
+
+    // This is not a backtransform diagnostics
+    bool const isBTD = false;
+    bool const isLastBTD = false;
```

Comment:
I am not sure what happens when writing multiple times to the **same** openPMD file **without** passing `isBTD=true`. Do you have any thoughts on this? 
Is there a possibility e.g. that the current code will erase any previously existing dataset in the file, at each successive write, if we don't pass `isBTD=true`. (Note: I did not have a detailed look at the internals of the function `DumpToFile`.)

Reply:
When the intervals is set, the scraped particle data is written out into separate files, a file each time the data is written out. This seems more robust, so for example the scraped particle data will be available as the simulation is running, and will be available if for some reason the simulation doesn't finish (it runs out of time or crashes for some other reason). This makes it natural to read in the data using as an `OpenPMDTimeSeries`.